pages/pipelines/group_step.md.erb

```diff
@@ -0,0 +1,78 @@
+# Group Step
+ 
+A *group* step groups up parallel jobs into logical groups and allows displaying them on the build page in the Buildkite UI.  
```

Comment:
So, I think we're slightly confusing two different things here. 

The *new* group step is unrelated to parallel steps. But it does *display* group steps in the same way as we already display group steps, in a compact way.